### ISSUE TITLE
Bug 58071131: [React Native Gallery -> Left Navigation]: Screen reader fails to announce the expand/collapse state of the navigation button.

### ISSUE THREAD

**Issue by vineethkuttan** [microsoft/react-native-gallery#612]
Build Details:
OS Version: Dev (OS Build 27871.1000)
App: React Native Gallery 
App Version: 1.0.18.0
React Native Windows Version: 0.79.0

AT: Screen Reader

Pre-requisites:

Turn on Narrator (Ctrl + Win + Enter).

Repro-Steps:
Launch the React Native Gallery app.
Navigate to 'Navigation' button (Expand/Collapse button) and listen to narrator announcement.
Actual Result:

Screen reader fails to announce the expand/collapse state of the navigation button. When user navigates/activates it narrator is announcing as "Navigation bar hamburger icon, button".

 

Expected Result:

Screen reader should announce the expand/collapse state of the navigation button. When user navigates to it narrator should announce as "Navigation bar, Expanded, button" or "Navigation bar, Collapsed, button".

 

User impact:

Screen Reader users will not be able to know that whether the pane is expanded or collapse state.

 

MAS Reference:

[MAS 4.1.2 – Name, Role, Value](https://aka.ms/MAS4.1.2)

Trap ID:
[1.4 - Uncomprehended Element](https://aka.ms/UncomprehendedElement)

"Have feedback on bugs logged, please tag bug as “A11yRCA” and add your feedback in the comment section.”

**Comment by chiaramooney** [comment]
This change should only be a JS side change (no Paper architecture specific knowledge needed). I think you'll just need to set accessibilityState for this button

Note: The source code for the Paper React Native Gallery begins at the root of the repo rather than inside the New Arch folder.

**Comment by vineethkuttan** [comment]
In Fabric, We should need to do this change.

**Comment by vineethkuttan** [comment]
The action pattern will set the button to expand or collapse state.
But if it is expanded, narrator should announce , like that is expanded.
If it is collapsed again, it should announce that it is collapsed.

Refer the attached screenshare video

**Comment by protikbiswas100** [comment]
Raise app level change https://github.com/microsoft/react-native-gallery/pull/647
